scala/scala_toolchain.bzl

```diff
@@ -3,12 +3,74 @@ load(
     _ScalacProvider = "ScalacProvider",
 )
 
+def _compute_dependency_mode(input_dependency_mode, input_plus_one_deps_mode):
+    if input_plus_one_deps_mode == "on":
+        return "plus-one"
+
+    if input_dependency_mode == "":
+        return "direct"
+
+    return input_dependency_mode
+
+def _compute_strict_deps_mode(input_strict_deps_mode, dependency_mode):
+    if dependency_mode == "direct":
+        return "off"
+    if input_strict_deps_mode == "default":
+        if dependency_mode == "transitive":
+            return "error"
+        else:
+            return "off"
+    return input_strict_deps_mode
+
+def _compute_dependency_tracking_method(input_dependency_tracking_method):
+    if input_dependency_tracking_method == "default":
+        return "high-level"
+    return input_dependency_tracking_method
+
 def _scala_toolchain_impl(ctx):
+    if ctx.attr.plus_one_deps_mode != "":
+        print(
+            "Setting plus_one_deps_mode on toolchain is deprecated." +
+            "Use 'dependency_mode = \"plus-one\"' instead",
+        )
+    if ctx.attr.dependency_mode != "" and ctx.attr.plus_one_deps_mode != "":
+        fail("Cannot set both dependency_mode and plus_one_deps_mode on toolchain")
+
+    if ctx.fragments.java.strict_java_deps != "default" and ctx.fragments.java.strict_java_deps != "off":
```

Comment:
this is for backward compatibility?

Reply:
also, does this mean that I'll be able to run with +1 only if I set strict_java_deps to warn/error? Sounds like a bug

Reply:
Yes for backwards compatibility.

Don't think it would mean you need to set strict_java_deps to run with +1. As this is saying, if strict_java_deps is set to warn/error then do this set of specific things which match previous behavior. And if not, then do the whole thing about reading from dependency_mode/plus_one_deps_mode to compute the dependency mode, etc, etc.

Reply:
You’re right. I misread.